node-hub/opencv-plot/main.py

```diff
@@ -113,50 +134,29 @@ def main():
                     "width": np.uint32(arrow_image["width"].as_py()),
                     "height": np.uint32(arrow_image["height"].as_py()),
                     "channels": np.uint8(arrow_image["channels"].as_py()),
-                    "data": arrow_image["data"].values.to_numpy().astype(np.uint8)
+                    "data": arrow_image["data"].values.to_numpy().astype(np.uint8),
                 }
 
-                plot.frame = np.reshape(image["data"], (image["height"], image["width"], image["channels"]))
+                plot.frame = np.reshape(
+                    image["data"], (image["height"], image["width"], image["channels"])
+                )
 
                 plot_frame(plot)
                 if not RUNNER_CI:
                     if cv2.waitKey(1) & 0xFF == ord("q"):
                         break
-                else:
-                    break  # break the loop for CI
-
             elif event_id == "bbox":
                 arrow_bbox = event["value"][0]
                 plot.bboxes = {
                     "bbox": arrow_bbox["bbox"].values.to_numpy().reshape(-1, 4),
                     "conf": arrow_bbox["conf"].values.to_numpy(),
                     "names": arrow_bbox["names"].values.to_numpy(zero_copy_only=False),
                 }
-
-                plot_frame(plot)
-                if not RUNNER_CI:
-                    if cv2.waitKey(1) & 0xFF == ord("q"):
-                        break
-                else:
-                    break  # break the loop for CI
             elif event_id == "text":
                 plot.text = event["value"][0].as_py()
-
-                plot_frame(plot)
-                if not RUNNER_CI:
-                    if cv2.waitKey(1) & 0xFF == ord("q"):
-                        break
-                else:
-                    break  # break the loop for CI
-
         elif event_type == "ERROR":
             raise Exception(event["error"])
 
-    node.send_output(
-        "end",
-        pa.array([0], type=pa.uint8())
-    )
-
```

Comment:
so when exiting the opencv-plot the dataflow doesn't end automatically, you need to CTRL+C, but it makes the camera fail:

```bash
(.env) enzo-le-van@enzo-le-van-PROX15-AMD:~/Documents/dora$ dora start examples/python-dataflow/dataflow.yml 
0190c986-829d-7910-b17e-f6d99e54adad
attaching to dataflow (use `--detach` to run in background)
INFO  plot: node finished successfully
^C

[ERROR]
Dataflow 0190c986-829d-7910-b17e-f6d99e54adad failed:

Node `camera` failed: node was killed by dora because it didn't react to a stop message in time (SIGKILL)
```

Reply:
Ah snap.

i don't have this problem on my computer:

```bash
~/D/w/d/e/python-dataflow ❯❯❯ dora start dataflow.yml                                                                                                                       (base) simplify-node-hub ✭
0190c9a5-2eab-7c89-973c-3b7e02c0c4bf
attaching to dataflow (use `--detach` to run in background)
^C%                                                                                                                                                                                                    
```

Reply:
You’re on release 0.3.5 or on the main branch ?

Reply:
I'm on main latest. But i know that the environment might influence the time for nodes to stop

Reply:
with the dora version the main latest I don't have this fail anymore